doc: nRF5340 DK getting started

wiba-nordic opened the pull request:
Move the getting started content from the infocenter to the NCS.

Also updates the content to align with more recent developments.

Signed-off-by: Wille Backman <wille.backman@nordicsemi.no>

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-10234
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

Avalei commented on doc/nrf/working_with_nrf/nrf53/nrf5340_gs.rst:
```diff
@@ -0,0 +1,109 @@
+After installing and starting the application, install the Programmer app.
+
+You must also install a terminal emulator, such as :ref:`PuTTY <putty>` or the nRF Terminal, which is part of the `nRF Connect for Visual Studio Code`_ extension.
+The steps detailed in :ref:`nrf5340_gs_connecting` use PuTTY, but any terminal emulator will work.
```
Do we want to highlight PuTTY steps instead of nRF Terminal? I thought we were strongly encouraging users to use the VSC extension.

wiba-nordic replied:
That could be good. The original version was using Termite, and I updated to PuTTY since that had documentation in the NCS but Termite didn't. But you are right, nRF Terminal is probably what we should be encouraging.

wiba-nordic replied:
I'll create a separate PR for that change since it'll be in the NRF52 Series file, to not muddy this PR too much.

Avalei commented on doc/nrf/working_with_nrf/nrf53/nrf5340_gs.rst:
```diff
@@ -0,0 +1,109 @@
+You must program and run a precompiled version of the :ref:`peripheral_uart` sample application on your development kit to test the functions.
+Download the precompiled version of the application from the `nRF5340 DK Downloads`_ page.
+
+After downloading the zip archive extract it to a folder of your choice.
```
```suggestion
After downloading the zip archive, extract it to a folder of your choice.
```

wiba-nordic commented:
Corrected use of "application" to "sample" in several places, and updated nRF Connect for Mobile iOS screenshot and instructions to match updated app.

Avalei commented on doc/nrf/working_with_nrf/nrf52/gs.rst:
```diff
@@ -60,26 +60,31 @@ The steps detailed in :ref:`nrf52_gs_connecting` use PuTTY, but any terminal emu
+
+.. program_dk_sample_start
+
+To program the precompiled sample sample to your development kit, complete the following steps:
```
```suggestion
To program the precompiled sample to your development kit, complete the following steps:
```
Looks like there's an accidental duplicate here.

wiba-nordic replied:
Good catch, used to say sample application and did the replacement too quickly.

umapraseeda commented on doc/nrf/working_with_nrf/nrf52/gs.rst:
```diff
@@ -96,34 +101,38 @@ To program the precompiled sample application to your development kit, complete
+.. program_dk_sample_end
+
+After you have programmed the sample to the DK, you can connect to it and test the functions.
+If you connect to the sample now, you can go directly to Step 2 of :ref:`nrf52_gs_connecting`.
```
```suggestion
If you connect to the sample now, you can go directly to Step 2 of :ref:`nrf52_gs_connecting` procedure.
```

umapraseeda commented on doc/nrf/working_with_nrf/nrf52/gs.rst:
```diff
@@ -175,24 +184,28 @@ To connect using CDC-UART, complete the following steps:
+
 The connection has now been established.
-If you test the application now, you can go directly to Step 2 of :ref:`nrf52_gs_testing`.
+If you test the sample now, you can go directly to Step 2 of :ref:`nrf52_gs_testing`.
```
```suggestion
If you test the sample now, you can go directly to Step 2 of :ref:`nrf52_gs_testing` procedure.
```

umapraseeda commented on doc/nrf/working_with_nrf/nrf53/nrf5340_gs.rst:
```diff
@@ -0,0 +1,109 @@
+========
+
+* nRF5340 DK
+* micro-USB 2.0 cable
```
```suggestion
* Micro-USB 2.0 cable
```

umapraseeda commented on doc/nrf/working_with_nrf/nrf53/nrf5340_gs.rst:
```diff
@@ -0,0 +1,109 @@
+Software
+========
+
+On your computer, one of the following operating systems:
```
I think we have already covered this in [this sentence](https://github.com/nrfconnect/sdk-nrf/pull/10234/files#diff-4306881ff8c514a522d0559898809372bcea37ad8afee76ea7db991a98f9b2b8R23).
```suggestion
```

wiba-nordic replied:
I kept the "On your computer:" to contrast with "On your mobile device:"

umapraseeda commented on doc/nrf/working_with_nrf/nrf53/nrf5340_gs.rst:
```diff
@@ -0,0 +1,109 @@
+* macOS X, latest version
+* Ubuntu Linux, latest Long Term Support (LTS) version
+
+On your mobile device, one of the following operating systems:
```
```suggestion
On your mobile device:
```

umapraseeda commented on doc/nrf/working_with_nrf/nrf53/nrf5340_gs.rst:
```diff
@@ -0,0 +1,109 @@
+
+.. _nrf5340_gs_installing_sample:
+
+Installing the sample
+*********************
```
```suggestion
Programming the sample
**********************
```

umapraseeda commented on doc/nrf/working_with_nrf/nrf53/nrf5340_gs.rst:
```diff
@@ -0,0 +1,109 @@
+   :start-after: program_dk_sample_start
+   :end-before: program_dk_sample_end
+
+After you have programmed the sample to the DK, you can connect to it and test the functions.
+If you connect to the sample now, you can go directly to Step 2 of :ref:`nrf5340_gs_connecting`.
```
Why don't we say something like, after programming the sample, you can connect to it and test the functions - followed by the subsequent sections? 
```suggestion
After you have programmed the sample to the DK, you can connect to it and test the functions.

```

wiba-nordic replied:
The sentence is there because the first step is to connect the DK. When doing the nRF52 GS, it initially said "If you connect to the sample now, you can skip Step 1", but there was feedback that it wasn't specific enough.

umapraseeda commented on doc/nrf/working_with_nrf/nrf53/nrf5340_gs.rst:
```diff
@@ -0,0 +1,109 @@
+   :end-before: uart_dk_connect_end
+
+The connection has now been established.
+If you test the sample now, you can go directly to Step 2 of :ref:`nrf5340_gs_testing`.
```
Not sure if we need this sentence here.
```suggestion
```

umapraseeda commented on doc/nrf/working_with_nrf/nrf53/nrf5340_gs.rst:
```diff
@@ -0,0 +1,109 @@
+You have now completed getting started with the nRF5340 DK.
+See the following links for where to go next:
+
+* :ref:`ug_nrf5340` for more advanced topics related to the nRF5340 DK.
```
```suggestion
* :ref:`ug_nrf5340` documentation for more advanced topics related to the nRF5340 DK.
```

umapraseeda commented on doc/nrf/working_with_nrf/nrf52/gs.rst:
```diff
@@ -96,34 +101,38 @@ To program the precompiled sample application to your development kit, complete
    Do not unplug or turn off the DK during this process.
 
 .. note::
    If you experience any problems during the process, press Ctrl+R (command+R on macOS) to restart the Programmer app, and try again.
```
```suggestion
   If you experience any problems during the process, press ``Ctrl+R`` (``command+R`` on macOS) to restart the Programmer app, and try again.
```

umapraseeda commented on doc/nrf/working_with_nrf/nrf52/gs.rst:
```diff
@@ -175,24 +184,28 @@ To connect using CDC-UART, complete the following steps:
+Testing the sample
+******************
+
+You can test the :ref:`peripheral_uart` sample on your DK using the `nRF Connect for Mobile`_ mobile application.
```
```suggestion
You can test the :ref:`peripheral_uart` sample on your DK using the `nRF Connect for Mobile`_ application.
```

umapraseeda commented on doc/nrf/working_with_nrf/nrf52/gs.rst:
```diff
@@ -175,24 +184,28 @@ To connect using CDC-UART, complete the following steps:
    .. group-tab:: Android
 
-      1. Connect the nRF52 Series DK to the computer with a micro-USB cable and turn on the DK.
+      1. Connect the |DK| to the computer with a micro-USB cable and turn on the DK.
```
Line 216 and 234: Should Nordic_UART_Service be in backticks?

umapraseeda commented on doc/nrf/working_with_nrf/nrf53/nrf5340_gs.rst:
```diff
@@ -0,0 +1,109 @@
+Testing the sample
+******************
+
+You can test the :ref:`peripheral_uart` sample on your |DK| using the `nRF Connect for Mobile`_ mobile application.
```
```suggestion
You can test the :ref:`peripheral_uart` sample on your |DK| using the `nRF Connect for Mobile`_ application.
```

umapraseeda commented on doc/nrf/working_with_nrf/nrf52/gs.rst:
```diff
@@ -60,26 +60,31 @@ The steps detailed in :ref:`nrf52_gs_connecting` use PuTTY, but any terminal emu
 
 On your mobile device, install the `nRF Connect for Mobile`_ mobile application from the corresponding application store.
```
```suggestion
On your mobile device, install the `nRF Connect for Mobile`_ application from the corresponding application store.
```

umapraseeda commented on doc/nrf/working_with_nrf/nrf53/nrf5340_gs.rst:
```diff
@@ -0,0 +1,109 @@
+You must also install a terminal emulator, such as :ref:`PuTTY <putty>` or the nRF Terminal, which is part of the `nRF Connect for Visual Studio Code`_ extension.
+The steps detailed in :ref:`nrf5340_gs_connecting` use PuTTY, but any terminal emulator will work.
+
+On your mobile device, install the `nRF Connect for Mobile`_ mobile application from the corresponding application store.
```
```suggestion
On your mobile device, install the `nRF Connect for Mobile`_ application from the corresponding application store.
```

wiba-nordic commented:
Waiting for the HEX file to be added to nRF5340 DK downloads section before asking for merge.
